components/net/blob_loader.rs

```diff
@@ -5,53 +5,92 @@
+             filemanager_chan: IpcSender<FileManagerThreadMsg>) {
+    let (chan, recv) = ipc::channel().unwrap();
+    if let Ok((id, origin, _fragment)) = parse_blob_url(&load_data.url.clone()) {
+        let id = SelectedFileId(id.simple().to_string());
+        let check_url_validity = true;
+        let msg = FileManagerThreadMsg::ReadFile(chan, id, check_url_validity, origin);
+        let _ = filemanager_chan.send(msg);
+
+        match recv.recv().unwrap() {
+            Ok(blob_buf) => {
+                let content_type: Mime = blob_buf.type_string.parse().unwrap_or(mime!(Text / Plain));
+                let charset = content_type.get_param(Attr::Charset);
+
+                let mut headers = Headers::new();
+
+                if let Some(name) = blob_buf.filename {
+                    let charset = charset.and_then(|c| c.as_str().parse().ok());
+                    headers.set(ContentDisposition {
+                        disposition: DispositionType::Inline,
+                        parameters: vec![
+                            DispositionParam::Filename(charset.unwrap_or(Charset::Us_Ascii),
+                                                       None, name.as_bytes().to_vec())
+                        ]
+                    });
+                }
+
+                headers.set(ContentType(content_type.clone()));
+                headers.set(ContentLength(blob_buf.size as u64));
+
+                let metadata = Metadata {
+                    final_url: load_data.url.clone(),
+                    content_type: Some(ContentType(content_type.clone())),
+                    charset: charset.map(|c| c.as_str().to_string()),
+                    headers: Some(headers),
+                    // https://w3c.github.io/FileAPI/#TwoHundredOK
+                    status: Some(RawStatus(200, "OK".into())),
+                    https_state: HttpsState::None,
+                    referrer: None,
+                };
 
-    headers.set(ContentType(content_type.clone()));
-    headers.set(ContentLength(blob_buf.size as u64));
-
-    let metadata = Metadata {
-        final_url: load_data.url.clone(),
-        content_type: Some(ContentType(content_type.clone())),
-        charset: charset.map(|c| c.as_str().to_string()),
-        headers: Some(headers),
-        // https://w3c.github.io/FileAPI/#TwoHundredOK
-        status: Some(RawStatus(200, "OK".into())),
-        https_state: HttpsState::None,
-        referrer: None
-    };
-
-    if let Ok(chan) =
-        start_sending_sniffed_opt(start_chan, metadata, classifier,
-                                  &blob_buf.bytes, load_data.context.clone()) {
-        let _ = chan.send(Done(Ok(())));
+                if let Ok(chan) =
+                    start_sending_sniffed_opt(start_chan, metadata, classifier,
+                                              &blob_buf.bytes, load_data.context.clone()) {
+                    let _ = chan.send(Payload(blob_buf.bytes));
```

Comment:
But `partial_body` is only used to classify the content as I can tell from the code :/